### ISSUE TITLE
Views

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-python#257]
I propose to add the `dj.View` class.  Essentially it wraps around a query and makes it look like a base relation or its own table.  But it does more than that.  For example, it can optimize the query by adding indexes on the queried tables.

For example, one of the most common queries for us is to get trials that correspond to a scan.  The query looks something like this:
```python
vis.Trial() * preprocess.Sync() & 'trial_idx between first_trial and last_trial'
```
We could now define the class
```python
@schema
class Trial(dj.View):
    definition = vis.Trial() * preprocess.Sync() & 'trial_idx between first_trial and last_trial'
```
and then use it in queries as
```python
(Trial() & key).fetch['flip_times']
```
Underneath, `dj.View` uses SQL's `CREATE VIEW` command to accommodate these queries.

**Comment by dimitri-yatsenko** [comment]
Views must be decorated with `@schema` since views are created within databases just like tables.

Creating  a view requires the `CREATE VIEW` privilege which we do not always provide.

**Comment by dimitri-yatsenko** [comment]
Views will not show in the ERD.

**Comment by dimitri-yatsenko** [comment]
implemented in #258 as `dj._View` for now.  As we identify all the use cases, we will rename to `dj.View`.

**Comment by dimitri-yatsenko** [comment]
This feature was removed in #271 as practical use of views had numerous issues and did not provide sufficient performance increase over derived relations.
